22.21.0: build failure with GCC 15

### Expected behavior
Build without failures with GCC 15 like it does with GCC 14.

### Actual behavior
Failure to build:

```
[...]
[ 37%] Building C object util/CMakeFiles/gvm_util_shared.dir/passwordbasedauthentication.c.o
cd /var/tmp/paludis/build/net-analyzer-gvm-libs-22.21.0/work/build/util && /usr/bin/x86_64-pc-linux-gnu-cc -DGVM_LIBS_VERSION=\"22.21.0\" -DHAVE_MQTT=1 -DREDIS_SOCKET_PATH=\"/tmp/redis.sock\" -Dgvm_util_shared_EXPORTS -I/usr/x86_64-pc-linux-gnu/include/glib-2.0 -I/usr/x86_64-pc-linux-gnu/lib/glib-2.0/include -I/usr/x86_64-pc-linux-gnu/include/libxml2 -march=native -O2 -pipe      -std=c11     -Wall     -Wextra     -Werror     -Wpedantic     -Wmissing-prototypes     -Wshadow     -Wsequence-point     -D_ISOC11_SOURCE     -D_DEFAULT_SOURCE -O3 -DNDEBUG -Wformat -Wformat-security -D_FORTIFY_SOURCE=2 -fstack-protector  -std=gnu11 -fPIC -fdiagnostics-color=always -MD -MT util/CMakeFiles/gvm_util_shared.dir/passwordbasedauthentication.c.o -MF CMakeFiles/gvm_util_shared.dir/passwordbasedauthentication.c.o.d -o CMakeFiles/gvm_util_shared.dir/passwordbasedauthentication.c.o -c /var/tmp/paludis/build/net-analyzer-gvm-libs-22.21.0/work/gvm-libs-22.21.0/util/passwordbasedauthentication.c
/var/tmp/paludis/build/net-analyzer-gvm-libs-22.21.0/work/gvm-libs-22.21.0/util/passwordbasedauthentication.c:45:3: error: initializer-string for array of 'char' truncates NUL terminator but destination lacks 'nonstring' attribute (65 chars into 64 available) [-Werror=unterminated-string-initialization]
   45 |   "./0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[...]
```

### Steps to reproduce

1. Use GCC 15
2. Try to build gvm-libs 22.21.0

### GVM versions

### Environment

**Operating system:** Exherbo Linux

**Installation method / source:** Source-based distribution

### Logfiles
Complete build log: [gvm-libs-22.21.0.log](https://github.com/user-attachments/files/20336985/gvm-libs-22.21.0.log)

--- a/util/passwordbasedauthentication.c
+++ b/util/passwordbasedauthentication.c
@@ -41,7 +41,7 @@ is_prefix_supported (const char *id)
 #ifndef EXTERNAL_CRYPT_GENSALT_R
 
 // used printables within salt
-const char ascii64[64] =
+const char ascii64[] =
   "./0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
 
 /* Tries to get BUFLEN random bytes into BUF; returns 0 on success. */

